recipes/log4cxx/all/conanfile.py

```diff
@@ -12,113 +16,150 @@ class Log4cxxConan(ConanFile):
-                  pattern=pattern)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
-    def _configure_cmake(self):
-        if not self._cmake:
-            self._cmake = CMake(self)
-            self._cmake.definitions["BUILD_TESTING"] = False
-            if self.settings.os == "Windows":
-                self._cmake.definitions["LOG4CXX_INSTALL_PDB"] = False
-            self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        apply_conandata_patches(self)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_TESTING"] = False
+        if Version(self.version) >= "1.0.0":
+            tc.variables["LOG4CXX_NETWORKING_SUPPORT"] = self.options.with_networking
+            tc.variables["LOG4CXX_MULTIPROCESS_ROLLING_FILE_APPENDER"] = self.options.with_multiprocess_rolling_file_appender
+            tc.variables["ENABLE_FMT_LAYOUT"] = self.options.with_fmt_layout
+        if Version(self.version) > "1.0.0" and self.settings.os != "Windows":
+            tc.variables["LOG4CXX_ENABLE_ESMTP"] = self.options.with_smtp_appender
+        tc.variables["LOG4CXX_ENABLE_ODBC"] = self.options.with_odbc_appender
+        tc.variables["LOG4CXX_CHAR"] = self.options.char_type
+        tc.variables["LOG4CXX_CHARSET"] = self.options.char_encoding
+        tc.variables["LOG4CXX_WCHAR_T"] = self.options.with_wchar_t
+        tc.variables["LOG4CXX_QT_SUPPORT"] = self.options.with_qt
+        if self.settings.os == "Windows":
+            tc.variables["LOG4CXX_INSTALL_PDB"] = False
+        tc.generate()
+
+        tc = CMakeDeps(self)
+        if Version(self.version) < "0.13.0":
+            tc.set_property("expat", "cmake_target_name", "EXPAT::EXPAT")
+        tc.generate()
 
     def build(self):
         self._patch_sources()
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
-        self.copy("NOTICE", src=self._source_subfolder, dst="licenses")
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "NOTICE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        if Version(self.version) < "1.0.0":
+            rmdir(self, os.path.join(self.package_folder, "share"))
+        if self.settings.os != "Windows":
+            rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
```

Comment:
```suggestion
        rmdir(self, os.path.join(self.package_folder, "share"))
        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
```
Let's simplify, rmdir does not raise an error when a folder does not exist.